helm/prometheus-rules/templates/alerting-rules/capo.rules.yml

```diff
@@ -0,0 +1,46 @@
+{{- if or (eq .Values.managementCluster.provider.kind "openstack") }}
+kind: PrometheusRule
+metadata:
+  labels: {{- include "labels.common" . | nindent 4}}
+  name: capi.rules
+  namespace: {{.Values.namespace}}
+spec:
+  groups:
+    - name: coredns
+      rules:
+        - alert: OpenStackMachineUnexpectedInstanceState
+          expr: capi_openstackmachine_status_instance_state{state!="ACTIVE"} > 0
+          for: 1h
+          labels:
+            area: kaas
+            cancel_if_outside_working_hours: {{include "workingHoursOnly" .}}
+            severity: notify
+            team: {{include "providerTeam" .}}
+            topic: managementcluster
+          annotations:
+            description: "{{`OpenStackMachine {{$labels.namespace}}/{{$labels.name}} instance state is {{$labels.state}}.`}}"
+
+        - alert: OpenStackMachineFailure
+          expr: capi_openstackmachine_status_failure{failure!="<nil>"}
```

Comment:
This looks fragile, but I guess there is no better option.

Reply:
For now there is no better option, but i've already have it on my list to make a PR (once the PR we currently based on) to support better `nil`-value behavior.